net/HttpRequest.hpp

```diff
@@ -1432,7 +1432,10 @@ class Session final : public ProtocolHandlerInterface
         while (!_response->done())
         {
             const auto now = std::chrono::steady_clock::now();
-            checkTimeout(now);
+            if (checkTimeout(now))
```

Comment:
When we timeout, the state of `_response` changes at line 1715, and the `_response->done()` at line 1432 returns true. If this is not the case, we should fix *that* issue.

Why do we care to break the loop here, is that to avoid the extra poll at 1442?

Reply:
I have to change this particular case in http::Session::syncRequestImpl()  - great catch!

Reply:
On a 2nd look, syncRequestImpl() can only return false if checkTimeout() returns true, i.e. closed the socket.
poller.poll() can't succeed here anymore and remaining should be zero.
Hence leaving the loop and function w/o further side-effects is OK IMHO.